File: 1672 - Richest Customer Wealth/MaximumWealth.js
// https://leetcode.com/problems/richest-customer-wealth/

function maximumWealth(accounts) {
  let max = Number.MIN_SAFE_INTEGER;

  for (let customer of accounts) {
    let wealth = 0;

    for (let bank of customer) {
      wealth += bank;
    }

    max = Math.max(wealth, max);
  }
  return max;
}

const accounts = [
  [1, 2, 3],
  [3, 2, 1],
  [4, 5, 6],
];

console.log(maximumWealth(accounts)); // 15

File: 1672 - Richest Customer Wealth/MaximumWealth.java
// https://leetcode.com/problems/richest-customer-wealth/

public class MaximumWealth {
    public static void main(String[] args) {
        int[][] accounts = {
                { 1, 2, 3 },
                { 3, 2, 1 },
                { 4, 5, 6 }
        };
        System.out.println(maximumWealth(accounts)); // 15
    }

    public static int maximumWealth(int[][] accounts) {
        int max = Integer.MIN_VALUE;

        for (int[] customer : accounts) {
            int wealth = 0;

            for (int bank : customer) {
                wealth += bank;
            }

            max = Math.max(wealth, max);
        }
        return max;
    }
}
